File: backend/catalog-service/core/brand/application/get_brands.py
from typing import List

from core.brand.domain.brand import Brand
from core.brand.domain.brand_repository import BrandRepository


class GetBrands:
    _brand_repository: BrandRepository

    def __init__(self, brand_repository: BrandRepository):
        self._brand_repository = brand_repository

    def execute(self) -> List[Brand]:
        return self._brand_repository.find_all()

File: frontend/web-app/src/core/brand/application/GetBrands.ts
import {ApiService} from "../../shared/services/ApiService";
import {Brand} from "../domain/Brand";

export class GetBrands {
    constructor(private apiClient: ApiService) {
    }

    execute(): Promise<Brand[]> {
        return this.apiClient.get<Brand[]>('/catalog/api/v1/brand/get_brands');
    }
}
